src/main/java/org/broadinstitute/hellbender/utils/test/IntegrationTestSpec.java

```diff
@@ -94,17 +92,20 @@ public void executeTest(final String name, CommandLineProgramTest test) throws I
             tmpFiles.add(fl);
         }
 
-        final String args = String.format(getArgs(), tmpFiles.toArray());
+        final String preFormattedArgs = getArgs();
+        final String formattedArgs = String.format(preFormattedArgs, tmpFiles.toArray());
         System.out.println(StringUtils.repeat('-', 80));
 
         if (expectsException()) {
             // this branch handles the case were we are testing that a walker will fail as expected
-            executeTest(name, test, null, null, tmpFiles, args, getExpectedException());
+            executeTest(name, test, null, null, tmpFiles, formattedArgs, getExpectedException());
         } else {
-            List<String> expectedFileNames = new ArrayList<>();
-            expectedFileNames.addAll(expectedFileNames());
+            final List<String> expectedFileNames = new ArrayList<>(expectedFileNames());
+            if (!expectedFileNames.isEmpty() && preFormattedArgs.equals(formattedArgs)){
+                throw new GATKException("Incorrect test specification - you're expecting " + expectedFileNames.size() + " file(s) the specified arguments do not contain the same number of \"%s\" placeholders");
```

Comment:
Add a unit test for `IntegrationTestSpec` proving that you get this exception when the spec is missing a placeholder.